features/step_definitions/means_report_steps.rb

```diff
@@ -24,19 +24,57 @@
   login_as @legal_aid_application.provider
 end
 
+Given("I have completed a non-passported employed application with enhanced bank statement uploads") do
+  @legal_aid_application = create(
+    :legal_aid_application,
+    :with_proceedings,
+    :with_employed_applicant,
+    :with_non_passported_state_machine,
+    :with_merits_statement_of_case,
+    :with_opponent,
+    :with_restrictions,
+    :with_incident,
+    :with_vehicle,
+    :with_transaction_period,
+    :with_extra_employment_information,
+    :with_other_assets_declaration,
+    :with_policy_disregards,
+    :with_fixed_offline_accounts,
+    :with_dependant,
+    :with_cfe_v5_result,
+    :with_chances_of_success,
+    :with_own_home_mortgaged,
+    :assessment_submitted,
+    property_value: rand(1...1_000_000.0).round(2),
+    outstanding_mortgage_amount: rand(1...1_000_000.0).round(2),
+    shared_ownership: LegalAidApplication::SHARED_OWNERSHIP_YES_REASONS.sample,
+    percentage_home: rand(1...99.0).round(2),
```

Comment:
I really don't think we should be writing non-deterministic tests.

We want our automated tests to be predictable and repeatable if they are to offer any value to us at all.

Adding randomness breaks this tenet and increases the likelihood of flaky tests.

I won't labour the point, but I think [Martin Fowler highlights some of these problems well](https://www.martinfowler.com/articles/nonDeterminism.html#WhyNon-deterministicTestsAreAProblem) —and I would encourage anyone who is skeptical to give that post a read.

Reply:
i agree, and have seen that post  - can tweak these easily